### DISCUSSION TITLE
React native boolean values from .env are loaded as strings

### DISCUSSION THREAD

**Discussion by cervebar** [goatandsheep/react-native-dotenv#477] [Q&A]
I have boolean value in my `.env` file written like this:

```
IS_DEVELOPMENT=true
```

And declared my `env.d.ts` like this:

```
declare module '@env' {
 ...
  export const IS_DEVELOPMENT: boolean
}

```

but when I use it in code:
```
import {IS_DEVELOPMENT} from '@env'
...
  console.log('IS_DEVELOPMENT', IS_DEVELOPMENT)
  console.log('IS_DEVELOPMENT', typeof IS_DEVELOPMENT)
  if (IS_DEVELOPMENT) {
    console.log('true value')
  } else {
    console.log('false value')
  }
```

not expected output is:
```
IS_DEVELOPMENT false
 LOG  IS_DEVELOPMENT string
 LOG  true value
```

In the previous version (`2.6.2`) we did the patch  with `dotenv-parse-variables` for this (but I generally don't prefer patched solutions).
```
diff --git a/node_modules/react-native-dotenv/index.js b/node_modules/react-native-dotenv/index.js
index 7210d37..2d63eca 100644
--- a/node_modules/react-native-dotenv/index.js
+++ b/node_modules/react-native-dotenv/index.js
@@ -1,5 +1,6 @@
 const {readFileSync} = require('fs')
 const dotenv = require('dotenv')
+const dotenvParseVariables = require('dotenv-parse-variables');
 
 function parseDotenvFile(path, verbose = false) {
   let content
@@ -54,6 +55,7 @@ module.exports = ({types: t}) => ({
       })
       this.env = process.env
     }
+    this.env = dotenvParseVariables(this.env)
   },
 
   visitor: {

```

How do you solve this? 
Can we somewhere say to the library that we actually want load env variables as booleans?
What are yours tips&tricks?

**Comment by goatandsheep** [comment]
My tips here would be to have a constants object and import from there

App.tsx
```typescript
import {IS_DEVELOPMENT} from './constants'
```

constants.tsx
```typescript
const constants = {
IS_DEVELOPMENT: process.env.IS_DEVELOPMENT === 'true' ? true : false
};
export default constants;
```

The counterintuitive part here is that the value you are checking for is _not_ the default. i.e. if no value is set, it will take the default. So if you don't put a value, the above will default to false. if you do `=== 'false' ? false : true`, then the code will default to true.

I will be covering this in my course I'll be launching this weekend.
